pykafka/balancedconsumer.py

```diff
@@ -504,18 +515,24 @@ def _check_held_partitions(self):
             self._rebalance()
 
     def _brokers_changed(self, brokers):
+        if not self._running:
+            return False  # disables the watcher
```

Comment:
That could definitely use a comment explanation, because it just looks wrong next to the succeeding lines.